src/idea.js

```diff
@@ -6,42 +6,44 @@ class Idea  {
     this.star = star;
     this.comments = [];
   }
-  saveToStorage(idea) {
+  saveToStorage() {
     var retrievedIdeas = JSON.parse(localStorage.getItem("ideas"));
-    if (!retrievedIdeas) {
-      retrievedIdeas = [];
-    } 
-    retrievedIdeas.push(idea);
-    localStorage.setItem("ideas", JSON.stringify(retrievedIdeas));
+    if (retrievedIdeas.length) {
+      for (var i = 0; i < retrievedIdeas.length; i++) {
+        //if retrievedIdeas contains any ideas and the card id already exists, update the card and then update storage
+        if (retrievedIdeas[i].id === this.id) {
+          retrievedIdeas[i] = this;
+          localStorage.setItem("ideas", JSON.stringify(retrievedIdeas));
+          return
+        }
+      }
+      //if retrievedIdeas contains any ideas and  the card id does not exist, push the card to the array of idea cards
+      retrievedIdeas.push(this);
+    } else {
+      //if retrievedIdeas is empty, push the idea card into the array
+      retrievedIdeas.push(this);
+    }
+      localStorage.setItem("ideas", JSON.stringify(retrievedIdeas));
   }
 
   deleteFromStorage() {
     var retrievedIdeas = JSON.parse(localStorage.getItem("ideas"));
+    //if retrievedIdeas is empty, exit the method
     if (!retrievedIdeas) {
       return;
-    } 
-
-    for (var i = 0; i < retrievedIdeas.length; i++) {
-      if (this.id === retrievedIdeas[i].id) {
-        retrievedIdeas.splice(i, 1);
+    } else {
```

Comment:
Else here is unnecessary - where there are no ideas we return and leave the function. If there are ideas we continue. Also, we had decided to take out the if(!retrievedIdeas) statement anyway so the whole function can look like this - 
```js
deleteFromStorage() {
    var retrievedIdeas = JSON.parse(localStorage.getItem("ideas"));	    
    for (var i = 0; i < retrievedIdeas.length; i++) {
      if (this.id === retrievedIdeas[i].id) {
        retrievedIdeas.splice(i, 1);
      }
    }
    localStorage.setItem("ideas", JSON.stringify(retrievedIdeas));
  }
```